src/Factory/RefundFactory.php

```diff
@@ -14,4 +14,12 @@ public function createWithData(string $orderNumber, int $unitId, int $amount, Re
     {
         return new Refund($orderNumber, $amount, $unitId, $type);
     }
+
+    /**
+     * @return object
+     */
+    public function createNew()
+    {
+        throw new \RuntimeException();
```

Comment:
the factory has to implement the `FactoryInterface::createNew` although its not possible to create a `Refund` without constructor parameter.

If the factory does not the ResouceController throws an exception.